report/metrics.go

```diff
@@ -52,28 +53,55 @@ type Sample struct {
 	Value     float64   `json:"value"`
 }
 
-var nilMetric = Metric{Samples: ps.NewList()}
+// MakeSingletonMetric makes a metric with a single value
+func MakeSingletonMetric(t time.Time, v float64) Metric {
```

Comment:
(Duplicated comment) 

It's not singleton in the OOP sense but in the mathematical sense https://en.wikipedia.org/wiki/Singleton_(mathematics) (I have borrowed this from Haskell).

Any other naming suggestions?